uniswap-v2/src/modules/14_map_liquidity_pool_snapshots_entity.rs

```diff
@@ -0,0 +1,500 @@
+    clock: Clock,
+    pools_store: StoreGetProto<Pool>,
+    pool_supply_store: StoreGetBigInt,
+    balances_store: StoreGetBigInt,
+    balances_deltas: Deltas<DeltaBigInt>,
+    pool_tvl_store: StoreGetBigDecimal,
+    cumulative_fields_store: StoreGetBigDecimal,
+    daily_and_hourly_fields_store: StoreGetBigDecimal,
+    volume_by_token_amount_store: StoreGetBigInt,
+    prices_store: StoreGetBigDecimal,
+) -> Result<EntityChanges, ()> {
+    let mut entity_changes: Vec<EntityChange> = vec![];
+    let timestamp = BigInt::from(clock.timestamp.unwrap().seconds);
+
+    for delta in balances_deltas.deltas.iter() {
+        if let Some(pool_address) = StoreKey::LatestTimestamp.get_pool(&delta.key) {
+            if delta.operation != store_delta::Operation::Create {
+                continue;
+            }
+
+            let delta_timestamp = delta.new_value.to_u64() as i64;
+
+            let day_id = utils::get_day_id(delta_timestamp) - BigInt::one();
+            let hour_id = utils::get_hour_id(delta_timestamp) - BigInt::one();
+
+            let block_number = balances_store
+                .get_at(delta.ordinal, StoreKey::LatestBlockNumber.unique_id())
+                .unwrap();
+
+            let pool: Pool =
+                pools_store.must_get_last(StoreKey::Pool.get_unique_pool_key(&pool_address));
+
+            entity_changes.push(create_liquidity_pool_daily_snapshot(
+                delta.ordinal,
+                &pool,
+                day_id,
+                &pool_supply_store,
+                &balances_store,
+                &pool_tvl_store,
+                &cumulative_fields_store,
+                &daily_and_hourly_fields_store,
+                &volume_by_token_amount_store,
+                &prices_store,
+                &block_number,
+                &timestamp,
+            ));
+
+            entity_changes.push(create_liquidity_pool_hourly_snapshot(
+                delta.ordinal,
+                &pool,
+                hour_id,
+                &pool_supply_store,
+                &balances_store,
+                &pool_tvl_store,
+                &cumulative_fields_store,
+                &daily_and_hourly_fields_store,
+                &volume_by_token_amount_store,
+                &block_number,
+                &timestamp,
+            ));
+        }
+    }
+
+    Ok(EntityChanges { entity_changes })
+}
+
+fn create_liquidity_pool_daily_snapshot(
+    ordinal: u64,
+    pool: &Pool,
+    day_id: BigInt,
+    pool_supply_store: &StoreGetBigInt,
+    balances_store: &StoreGetBigInt,
+    pool_tvl_store: &StoreGetBigDecimal,
+    cumulative_fields_store: &StoreGetBigDecimal,
+    daily_and_hourly_fields_store: &StoreGetBigDecimal,
+    volume_by_token_amount_store: &StoreGetBigInt,
+    prices_store: &StoreGetBigDecimal,
+    block_number: &BigInt,
+    timestamp: &BigInt,
+) -> EntityChange {
+    let id = [pool.clone().address, day_id.clone().to_string()].join("-");
```

Comment:
Haven't checked to see if it works but try like: `[pool.address.as_str(), day_id.to_string().as_str()].join("-");` instead